src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs

```diff
@@ -1025,7 +1026,9 @@ public async Task TcpReceiveSendGetsCanceledByDispose(bool receiveOrSend, bool i
                             var buffer = new ArraySegment<byte>(new byte[4096]);
                             while (true)
                             {
-                                SendAsync(socket1, buffer).GetAwaiter().GetResult();
+                                SendAsync(socket1, buffer)
+                                    .WaitAsync(TimeSpan.FromMilliseconds(TestSettings.PassingTestTimeout))
+                                    .GetAwaiter().GetResult();
```

Comment:
This could also just be:
```C#
.Wait(TimeSpan.FromMilliseconds(TestSettings.PassingTestTimeout));
```
If that fails it'll end up throwing an AggregateException, but I don't see anything paying attention to the exact exception type anyway...